### PR TITLE
pybind: configurable cephfs_vol_client prefix.

### PR DESCRIPTION
**Pull request by xiaoxichen**
Which is useful if we would like to have multiple backends
in manila and potentially set different layout based on prefix.

Signed-off-by: Xiaoxi Chen <xiaoxchen@ebay.com>

### CODE REVIEW

**Comment by gregsfortytwo** [comment]
ping @jcsp, unless @ajarr wants to check it out

File: src/pybind/ceph_volume_client.py [hunk 1]
```diff
@@ -203,17 +203,19 @@ class CephFSVolumeClient(object):
...
 
-    def __init__(self, auth_id, conf_path, cluster_name):
+    def __init__(self, auth_id, conf_path, cluster_name, volume_prefix=None, pool_ns_prefix=None):
         self.fs = None
         self.rados = None
         self.connected = False
         self.conf_path = conf_path
         self.cluster_name = cluster_name
         self.auth_id = auth_id
+        self.volume_prefix = volume_prefix if volume_prefix else DEFAULT_VOL_PREFIX
```
Comment (ajarr) [inline src/pybind/ceph_volume_client.py]: s/DEFAULT_VOL_PREFIX/self.DEFAULT_VOL_PREFIX/

File: src/pybind/ceph_volume_client.py [hunk 2]
```diff
@@ -203,17 +203,19 @@ class CephFSVolumeClient(object):
...
+        self.pool_ns_prefix = pool_ns_prefix if pool_ns_prefix else DEFAULT_NS_PREFIX
```
Comment (ajarr) [inline src/pybind/ceph_volume_client.py]: s/DEFAULT_NS_PREFIX/self.DEFAULT_NS_PREFIX/

**Comment by ajarr** [comment]
Maybe it would be helpful to refer to tracker issue that this commit "fixes/partially fixes"
http://tracker.ceph.com/issues/15417
I'm not sure about commit tags for patches adding new features.

**Comment by ajarr** [comment]
Tested by https://github.com/ceph/ceph-qa-suite/pull/975

**Comment by xiaoxichen** [comment]
@ajarr ,comments addressed.

**Comment by ajarr** [comment]
@xiaoxichen , the tracker issue number in the commit message is incorrect. And please mention the link there http://tracker.ceph.com/issues/15417 instead of just the number.

**Comment by xiaoxichen** [comment]
@ajarr , updated the number. Actually in practice, we always  only put the number in commit message.

**Comment by ajarr** [comment]
@xiaoxichen , not really.  Checkout, https://github.com/ceph/ceph/commit/65770059cb319da9ffbc2866df1ffa8b35267614

**Comment by ajarr** [comment]
Code looks good to me.

**Comment by jcsp** [comment]
@xiaoxichen merged, thanks!  The number vs. URL thing isn't a big deal, but we are switching to using URLs because the tracker numbers were colliding with github pull request numbers; future patches should use URL.

**Pull request merged**